var/spack/repos/builtin/packages/chameleon/package.py

```diff
@@ -0,0 +1,86 @@
+# Copyright 2013-2022 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+
+class Chameleon(CMakePackage, CudaPackage):
+    """Dense Linear Algebra for Scalable Multi-core Architectures and GPGPUs"""
+    homepage = "https://gitlab.inria.fr/solverstack/chameleon"
+    url      = "https://gitlab.inria.fr/solverstack/chameleon/uploads/b299d6037d7636c6be16108c89bc2aab/chameleon-1.1.0.tar.gz"
+    git      = "https://gitlab.inria.fr/solverstack/chameleon.git"
+
+    version('master', branch='master', submodules=True)
+    version('1.1.0', 'e64d0438dfaf5effb3740e53f3ab017d12744b85a138b2ef702a81df559126df')
+
+    # cmake's specific
+    variant('shared', default=True, description='Build chameleon as a shared library')
+
+    # chameleon's specific
+    variant(
+        'runtime', default='starpu', description='Runtime support',
+        values=('openmp', 'starpu'), multi=False
+    )
+    variant('mpi', default=True, description='Enable MPI')
+    variant('cuda', default=False, description='Enable CUDA')
+    variant('fxt', default=False, description='Enable FxT tracing support through StarPU')
+    variant('simgrid', default=False, description='Enable simulation mode through StarPU+SimGrid')
+
+    # dependencies
+    depends_on("pkg-config", type='build')
+    depends_on("blas", when='~simgrid')
+    depends_on("lapack", when='~simgrid')
+    depends_on("starpu", when='runtime=starpu')
+    depends_on("starpu~mpi", when='~mpi runtime=starpu')
+    depends_on("starpu+cuda", when='+cuda ~simgrid runtime=starpu')
+    depends_on("starpu+fxt", when='+fxt runtime=starpu')
+    depends_on("starpu+simgrid", when='+simgrid runtime=starpu')
+    depends_on("starpu+mpi~shared+simgrid", when='+mpi +simgrid runtime=starpu')
```

Comment:
If it makes sense, you can group common requirements together like:
```suggestion
    with when("runtime=starpu"):
        depends_on("starpu")
        depends_on("starpu~mpi", when='~mpi')
        depends_on("starpu+mpi", when='+mpi')
        depends_on("starpu+cuda", when='+cuda ~simgrid')
        depends_on("starpu+fxt", when='+fxt')
        depends_on("starpu+simgrid", when='+simgrid')
```